File: Cryptography/wasif-sir/caesarCipher.go
package main

import "fmt"

func cs(text string, key int) string {
	result := ""
	for _, ch := range text {
		if (ch >= 'A' && ch <= 'Z') || (ch >= 'a' && ch <= 'z') {
			is_upper := (ch >= 'A' && ch <= 'Z')
			var b rune
			if is_upper {
				b = 'A'
			} else {
				b = 'a'
			}
			var keyed_char = (ch-b+rune(key)+26)%26 + b
			result += string(keyed_char)
		} else {
			result += string(ch)
		}
	}
	return result
}
func csd(text string, key int) string {
	return cs(text, -key)
}

func task1() {
	var input rune
	fmt.Scanf("%c", &input)
	fmt.Println(string(input))
	fmt.Println(string(input - ('a' - 'A')))
}
func task2() {
	var input string
	var key int
	fmt.Scanf("%s", &input)
	fmt.Scanf("%d", &key)
	encrypted := cs(input, key)
	decrypted := csd(encrypted, key)
	print("Encrypted: " + encrypted + "\n")
	println("Decrypted: " + decrypted + "\n")
}

File: Cryptography/wasif-sir/caesarCipher.cpp
#include<bits/stdc++.h>
using namespace std;

string encrypt(const string& text, int key) {
    string result = "";
    for (char ch : text) {
        if (isalpha(ch)) {
            bool is_upper = isupper(ch);
            char keyed_char = ((ch - (is_upper ? 'A' : 'a') + key) % 26 + 26) % 26 + (is_upper ? 'A' : 'a');
            result += keyed_char;
        } else {
            result += ch;
        }
    }
    return result;
}

string decrypt(const string& ciphertext, int key) {
    return encrypt(ciphertext, -key);
}

int main() {
    string plaintext;
    cout<<"Write a message: ";
    getline(cin, plaintext);
    cout<<"Enter a secret key: ";
    int key_value;
    cin>>key_value;
    string encrypted_text = encrypt(plaintext, key_value);
    string decrypted_text = decrypt(encrypted_text, key_value);

    cout << "Original Text: " << plaintext << endl;
    cout << "Encrypted Text: " << encrypted_text << endl;
    cout << "Decrypted Text: " << decrypted_text << endl;

    return 0;
}